logs/logs.go

```diff
@@ -106,7 +108,15 @@ func (l *LogAgent) Run(ctx context.Context) {
 						log.Printf("E! [logagent] Failed to find destination %v for log source %v/%v(%v) ", dname, src.Group(), src.Stream(), src.Description())
 						continue
 					}
-					dest := backend.CreateDest(src.Group(), src.Stream(), src.Retention())
+					retention := src.Retention()
+					if retention > 0 && l.retentionAlreadySet[src.Group()] {
+						log.Printf("D! [logagent] Retention already set for log group %v/%v retention:%v) ", src.Group(), src.Stream(), src.Retention())
```

Comment:
Minor thing - I think `%v` is a good catch-all, but we know the types of these values so we can be more direct:
```suggestion
						log.Printf("D! [logagent] Retention already set for log group %s/%s retention:%d) ", src.Group(), src.Stream(), src.Retention())
```